grobian@gentoo.org: fix compilation on macOS using GCC

compiler says it all: include memory if you use unique_ptr

--- a/Source/cmMachO.h.orig	2020-11-24 14:50:26.000000000 +0100
+++ b/Source/cmMachO.h	2020-11-28 16:05:11.509620124 +0100
@@ -6,6 +6,7 @@
 
 #include <iosfwd>
 #include <string>
+#include <memory>
 
 #if !defined(CMAKE_USE_MACH_PARSER)
 #  error "This file may be included only if CMAKE_USE_MACH_PARSER is enabled."
